stellarphot/settings/tests/test_astropy_pydantic.py

```diff
@@ -0,0 +1,242 @@
+import json
+from typing import Annotated
+
+import numpy as np
+import pytest
+from astropy.coordinates import SkyCoord
+from astropy.time import Time
+from astropy.units import Quantity, Unit, get_physical_type
+from pydantic import BaseModel, ValidationError
+
+from stellarphot.settings.astropy_pydantic import (
+    AstropyValidator,
+    EquivalentTo,
+    QuantityType,
+    UnitType,
+    WithPhysicalType,
+    serialize_astropy_type,
+)
+
+
+class _UnitModel(BaseModel):
+    # Dummy class for testing
+    unit: UnitType
+
+
+class _QuantityModel(BaseModel):
+    # Dummy class for testing
+    quantity: QuantityType
+
+
+@pytest.mark.parametrize(
+    "init",
+    [
+        Unit("m"),
+        1,
```

Comment:
What do you mean by "float with a unit"?  I think `5 * Unit('m')` is already guaranteed to be a `Unit` so it is covered by the `Unit("m")` case.

Reply:
> What do you mean by "float with a unit"? I think `5 * Unit('m')` is already guaranteed to be a `Unit` so it is covered by the `Unit("m")` case.

Oh, I still have this idea that 5 * Unit('m') is different from 5.0*Unit('m') since Astropy values with units I thought got stored as numerical type + unit information.  If I am incorrect, forget it.